### DISCUSSION TITLE
Live stream from socket.io transcription not working

### DISCUSSION THREAD

**Discussion by sagardesai90** [deepgram/community#529] [General help]
I've been trying to get live streaming transcription to work, but haven't been able to figure out what's not working. Here's a couple of the code snippets that're responsible for handling audio caption, encoding, as well as node-side handling of audio blob before sending to deepgram.

<img width="1684" alt="Screenshot 2024-01-12 at 4 12 46 PM" src="https://github.com/deepgram/community/assets/24217185/d9376d26-0fe8-4bf9-a4ad-12a3e6a60e9e">

This is another part of server.js which is responsible for receiving an 'audio-stream' and sending that to deepgram.

<img width="880" alt="Screenshot 2024-01-12 at 4 19 21 PM" src="https://github.com/deepgram/community/assets/24217185/00ab1f8f-819c-4919-aad5-5e3a2f3c5528">

The terminal output looks like this, where there is actually a transcript for a split second, but then it stops, despite deepgram being sent audio after that. I'd expected to see a continual transcript from deepgram but didn't see anything.

<img width="1193" alt="Screenshot 2024-01-12 at 4 10 50 PM" src="https://github.com/deepgram/community/assets/24217185/413af7ec-c718-4f20-bf99-2ba3ba878beb">

Logs on Deepgram dashboard look like this:

<img width="1508" alt="Screenshot 2024-01-12 at 4 23 38 PM" src="https://github.com/deepgram/community/assets/24217185/236e7747-6ffa-4378-a6cc-dc947ac5aaaf">

Request ID: 42eda1e8-3d97-4ced-8260-7ae72643ccab

Is there a reason why the transcription isn't occurring properly? I've been looking through documentation to see what it can be but haven't found a solution yet. Thanks!

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by SandraRodgers** [comment]
Hi @sagardesai90,

I'm not sure why your live transcription code isn't working, but I think you could take a look at this example to help you debug:  https://github.com/deepgram-starters/live-node-starter/tree/socket-io

I created a separate branch that shows it working with socket.io. Be sure you are on that branch when you test it out.

If you still can't figure it out, feel free to send me a link to your project and I can test out your code to try and troubleshoot the issue.

Sandra

**Reply by sagardesai90** [reply]
Hey @SandraRodgers, 

Thanks for the reply. I looked at the documentation you referred to on the socket-io branch, and it seems like I'm using the same code for the most part. I do see a transcript 4 times like you see in the terminal output in the original post, but there isn't an actual transcript. Furthermore, I'm pretty sure there is audio data being sent to deepgram, as shown in the logs on deepgram. 

My repo is private, so I'd have to add you as a collaborator temporarily, does that work?

Sagar

**Reply by SandraRodgers** [reply]
Hi Sagar,

I'm wondering, when you see the `data.channel` object in the terminal (what comes in as a response for your `console.log("socket: transcript sent to client", data.channel);`, are you actually speaking into the microphone at that time? Because you should see an empty transcript like that if the microphone is open but you aren't speaking. The livestream will continue to send back an empty string continuously, even if nothing was said. This would be normal behavior.

But I'm not sure why it shows `deepgram: disconnected` so that is something I'd want to know more about or try to figure out if it's unexpected behavior.

If you feel comfortable adding me as a temporary collaborator, you could do that. I will troubleshoot it tomorrow morning US time since my workday today is almost finished.

Sandra

**Reply by sagardesai90** [reply]
<img width="1164" alt="Screenshot 2024-01-16 at 3 26 54 PM" src="https://github.com/deepgram/community/assets/24217185/ec2f82cc-b12f-4a32-861f-ac64017c773d">

This is the terminal output when I start the node server, and the audio starts coming in from client. Even if there is speech during the start, the transcript shows up as empty. Secondly, there are only 4 times (over a split second) that a transcript response comes in from deepgram. I'm not sure why the connection gets disconnected right after that, on a consistent basis. I've tried looking at my code and there isn't anything thats stands out. Would be happy to add you to the project so you can take a closer look at this issue, thanks again! 

Just sent the invite, lmk if you get it!

**Reply by SandraRodgers** [reply]
I've received the invite and I'll take a look at the project today. Thanks for your patience. :-)

**Reply by SandraRodgers** [reply]
@sagardesai90 Can you tell me which branch I should look at? I don't see Deepgram being used on the master branch.

**Reply by sagardesai90** [reply]
I need to push some changes, doing that now!

**Reply by sagardesai90** [reply]
Added a new branch 'deepgram' with the changes, thanks for taking a look and let me know if you need anything else. I can also get on a zoom call if you need me to explain anything.

**Reply by SandraRodgers** [reply]
I pulled in your code, switched to the deepgram branch, ran `npm install` and ran `npm run start`. It says this in the terminal
```
{} sessions
server is running on port 8000
```
When I go to `http://localhost:8000/`, I just see an error `Cannot GET /`. Could you advise me how I can run the app locally?

**Reply by sagardesai90** [reply]
Thanks for the update Sandra. You have to run npm install in the client folder and have both running. 

1-npm install + nodemon server.js in root

2-npm install + npm run start in /client folder

I also hard coded one part of the code for now, so be sure to go to http://localhost:3000/box/TrHmQ, thats the sessionID I used for now. 

Once you have both the node and client side up and running, it should start sending audio data to the node server, then to deepgram. I can also get on a call if you have any other questions/blockers.
